specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/datafactory.json

```diff
@@ -2854,6 +2902,56 @@
           "type": "object"
         }
       },
+      "FactoryVSTSConfiguration": {
+        "description": "Factory's VSTS repo information.",
+        "properties": {
+          "accountName": {
+            "description": "VSTS account name.",
+            "type": "string"
+          },
+          "projectName": {
+            "description": "VSTS project name.",
+            "type": "string"
+          },
+          "repositoryName": {
+            "description": "VSTS repository name.",
+            "type": "string"
+          },
+          "collaborationBranch": {
+            "description": "VSTS collaboration branch.",
+            "type": "string"
+          },
+          "rootFolder": {
+            "description": "VSTS root folder.",
+            "type": "string"
+          },
+          "lastCommitId": {
+            "description": "VSTS last commit id.",
+            "type": "string"
+          },
+          "tenantId": {
+            "description": "VSTS tenant id.",
+            "type": "string"
+          }
+        }
+      },
+      "FactoryRepoUpdate": {
+        "description": "Factory's VSTS repo information.",
+        "properties": {
+          "factoryResourceId": {
+            "description": "The factory resource id.",
+            "type": "string"
+          },
+          "resourceGroupName": {

```

Comment:
I'd ref this definition instead, in order to get client-side validation:
https://github.com/Azure/azure-rest-api-specs/pull/2866/files#diff-bf6d272fbfeeed92d1fc24a2bb8706a8R3781

Reply:
Chatted offline with @frankycrm14 about this, it looks like the defintion I mentioned is in the parameters section. For that reason, the real fix would be to refactor most of this into the definitions section, and reference that from the parameter and anywhere else it's used. However, that would only buy us a little client-side validation, and wouldn't be a breaking change to fix in the future. For those reasons, I'm not going to block this PR on this kinda comment.

All that remains before I merge is ensuring that none of the linter errors being reported were introduced by this PR.